src/Views/VPNPage.vala

```diff
@@ -274,50 +218,123 @@ public class Network.VPNPage : Network.Widgets.Page {
         return item;
     }
 
-    private VPNMenuItem? get_previous_menu_item () {
-        var children = vpn_list.get_children ();
-        if (children.length () == 0) {
-            return blank_item;
+    private void update_active_connections () {
+        if (active_connections == null) {
+            active_connections = new Gee.ArrayList<NM.VpnConnection> ();
```

Comment:
This check won't do anything useful being here, since you call `update_active_connections` that's called by `update` which in turn is called by the constructor. And at that time `active_connections` is null anyway.

It would be better if `active_connections` was initialized always in the constructor and be always initialized through the lifetime of the instance. This method should just clear the list.